adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/UserCredential.cs

```diff
@@ -26,50 +26,57 @@
 //------------------------------------------------------------------------------
 
 using Microsoft.IdentityModel.Clients.ActiveDirectory.Internal;
+using Core = Microsoft.Identity.Core;
 
 namespace Microsoft.IdentityModel.Clients.ActiveDirectory
 {
-    internal enum UserAuthType
-    {
-        IntegratedAuth,
-        UsernamePassword
-    }
+    using UserAuthType = Core.UserAuthType;
+
 
     /// <summary>
     /// Credential used for integrated authentication on domain-joined machines.
     /// </summary>
     public class UserCredential
     {
+        internal Core.UserCredential _UserCredential;
```

Comment:
The short answer is, the underlying `Core.WsTrust.WsTrustRequest.SendRequestAsync(..., Core.UserCredential userCredential, ...)` will only accept a `Core.UserCredential` object defined in core space. And here in `Adal.UserCredential`, we use composition (rather than inheritance), so we will have to provide a way to reference that inner property. The current approach is a workaround that satisfies the other constraints.

---
In reply to: [206765835](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1118#discussion_r206765835) [](ancestors = 206765835)